nestkernel/connection_manager.cpp

```diff
@@ -1467,7 +1467,12 @@ nest::ConnectionManager::connection_required( Node*& source, Node*& target, thre
   // currently prohibit connections from devices to global receivers.
   else
   {
-    if ( source->has_proxies() )
+
+    const index spike_gen_id = kernel().model_manager.get_node_model_id( "spike_generator" );
+    const index src_model_id = kernel().model_manager.get_node_model_id( source->get_name() );
+    const auto source_is_spike_gen = spike_gen_id == src_model_id;
```

Comment:
Here `"spike_generator"` is converted to a `Name` object before getting the model ID for every connection going into this clause. Isn't it better to store the `spike_gen_id` value in ConnectionManager (and set it in `ConnectionManager::initialize()`) so we only have to get it once?

And you can get the model ID of the source node directly with `source->get_model_id()`.

Reply:
Additionally, this does not work for models that came into existence by being `CopyModel()'`d from the original `spike_generator` and makes the NEST kernel dependent on the availability of a model called `spike_generator`.

Moreover this adds to the connection time of each device to support this special case...

Reply:
I agree with you both, and another procedure is needed to distinguish nodes with the properties we want for `spike_generator`. For the record, it is also not possible to retrieve `spike_generator`s model id in `ConnectionManager::initialize` because the models are registered by `ModelsModule::init` at a later stage.